transports/transports-deciding.md

```diff
@@ -0,0 +1,196 @@
+It has benefits for learning how to work with NServiceBus. Instead of requiring additional technology installed, it will work straight out of the box. This is achieved by the fact that this transport stores messages as files on disk which can be read by other endpoints.
+
+
+
+## MSMQ Transport
+
+The MSMQ transport uses native Windows queuing mechanism, MSMQ, to deliver messages. MSMQ is a distributed system that consists of multiple processes, one on each machine. The client always interacts with a process running locally and the messages are forwarded in the background.
+
+#### Advantages
+
+- A built-in component of Windows operating system
+- Supports distributed transactions allowing atomic message receive and data manipulation through [Microsoft Distributed Transaction Coordinator (MSDTC)](https://msdn.microsoft.com/en-us/library/ms684146.aspx)
+- Provides a store and forward mechanism which allows endpoints to send messages even though the destination endpoint might be temporarily unreachable due to network issues.
+
+#### Disadvantages
+
+- Does not offer a native publish-subscribe mechanism, therefore it requires NServiceBus persistence to be configured for storing event subscriptions. [Explicit routing for publish/subscribe](/nservicebus/messaging/routing.md#event-routing-message-driven) must also be specified.
+- Many organizations don't have the same level of operational expertise with MSMQ that they do with other technologies (e.g. SQL Server), so it may require additional training.
+- Scaling out compute-intensive workload require setting up the distributor or sender-side distribution of messages to destination queues. This is harder than with the competing consumer pattern that other transports support.
+
+#### When would you select this transport
+
+- If you need store & forward for a higher reliability that your queuing technology is always available.
+- If you’re running a Windows environment on-premise and don’t want to spend additionally on licenses and/or training.
+- If you are able to coop distributed transactions and want the reliability it adds.
+
+
+
+## Azure Service Bus Transport
+
+Azure provides multiple messaging technologies and one of the most advanced and reliable ones is Azure Service Bus.
+
+Advantages
+
+- No need to maintain infrastructure or have deep knowledge of Azure Service Bus as Microsoft and NServiceBus take care of almost everything.
+- Ability to scale in both features and price range.
+- AMQP based implementation
+- Supports messaging transactions unlike other messaging services on Azure
+- Up to 1MB message size
+- Levering native capabilities of the transport instead of taxing an endpoint
+
+## Disadvantages
+
+- Requires monitoring of costs, although NServiceBus supports tweaking number of transactions to Azure.
+- No on-premise option for development or testing
+- Processing a message has to complete within 5 minutes
+- Depends on TCP and might require opening additional ports in the firewall
```

Comment:
It also support websockets with URI scheme `wss` and port 443.

Reply:
While AMQP and WebSockets both rely on TCP, the difference is for those that cannot have custom ports open. WebSockets use a standard port 443 which can be used when the port(s) required for AMQP could be blocked.

Reply:
The current text says (emphasis mine):

> Relies on TCP, which _may_ require opening additional ports in a firewall.

So I think we are OK.